pkg/compact/schedule_delete.go

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package compact
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/oklog/ulid"
+	"github.com/pkg/errors"
+	"github.com/thanos-io/thanos/pkg/block"
+	"github.com/thanos-io/thanos/pkg/objstore"
+	"github.com/thanos-io/thanos/pkg/runutil"
+)
+
+// DebugCompactorMetas is folder to store compactor metadata
+// about when the block is scheduled to be deleted.
+const DebugCompactorMetas = "debug/compactor-metas"
```

Comment:
Why "debug" prefix? I assume this will be part of normal compactor flow. Perhaps something like "delete-schedule" would make more sense.

Reply:
removed this folder, now we are uploading as `compactor-meta.json` file inside block

Reply:
I like the file inside block approach much better. Thanks!